pkg/client/client.go

```diff
@@ -47,29 +45,34 @@ func NewWithBasicAuth(agentMetadata *api.AgentMetadata, authToken, baseURL strin
 }
 
 // New creates a new client that uses OAuth2.
-func New(agentMetadata *api.AgentMetadata, userID, userSecret, baseURL string) (*PreflightClient, error) {
-	if userID == "" || userSecret == "" {
+func New(agentMetadata *api.AgentMetadata, credentials *Credentials, baseURL string) (*PreflightClient, error) {
+	if credentials == nil || credentials.UserID == "" || credentials.UserSecret == "" {
```

Comment:
Is it worth validating the userID and secret are set earlier than this function? (see note above)

Reply:
Shouldn't be nil here because of https://github.com/jetstack/preflight/pull/166/files#diff-d1aa418498ff92dc3def120d994c0379R98 right?

Reply:
Oh, I forgot we had a validation method for `Credentials`.

In that case, I am going to change this so `New` checks `credentials` is not `nil` and it is indeed valid credentials.

Having `New` do that sanity check is a good thing, as this package could be used somewhere else.

Reply:
:+1: sounds good.